### PR TITLE
Use --depth to speed up pristine submodule cloning

### PR DESCRIPTION
**Pull request by aidanhs**
In addition, remove all the unused cache stuff

r? @Mark-Simulacrum

### CODE REVIEW

**Comment by aidanhs** [comment]
I expect this to save us 5-7min on OSX builds.

File: src/ci/init_repo.sh [hunk 1]
```diff
@@ -36,47 +33,22 @@ if [ ! -d "$CACHE_DIR" ]; then
...
 travis_fold start update_cache
 travis_time_start
 
 # Update the cache (a pristine copy of the rust source master)
 if [ ! -d "$cache_src_dir/.git" ]; then
     retry sh -c "rm -rf $cache_src_dir && mkdir -p $cache_src_dir && \
-        git clone https://github.com/rust-lang/rust.git $cache_src_dir"
+        git clone --depth 1 https://github.com/rust-lang/rust.git $cache_src_dir"
 fi
 retry sh -c "cd $cache_src_dir && git reset --hard && git pull"
```
Comment (Mark-Simulacrum) [inline src/ci/init_repo.sh]: `git pull` here I think will make us no longer have a depth of 1 anyway, but I'm not sure. https://stackoverflow.com/questions/19352894/how-to-git-fetch-efficiently-from-a-shallow-clone

Comment (aidanhs) [inline src/ci/init_repo.sh]: Oops, I have some other changes I didn't push.

**Comment by Mark-Simulacrum** [comment]
I'll wait for Travis, but looks good to me overall.

**Comment by Mark-Simulacrum** [comment]
@bors r+ p=1

Setting higher priority since this helps with OS X timeouts (at least a little).

Looks like this passed the cache stage in Travis.

**Pull request merged**